### PR TITLE
(FACT-2687) --trace shows backtrace for core facts

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

**Comment by Filipovici-Andrei** [comment]
jenkins please test this

File: lib/facter/framework/logging/logger.rb [hunk 1]
```diff
@@ -3,10 +3,11 @@
 require 'logger'
 
 module Facter
-  RED = 31
-  GREEN = 32
-  YELLOW = 33
-  CYAN = 36
+  RED     = "\e[31m"
+  GREEN   = "\e[32m"
+  YELLOW  = "\e[33m"
```
Comment (oanatmaria) [inline lib/facter/framework/logging/logger.rb:8]: \e[0;{COLOR_CODE}m https://www.cyberciti.biz/faq/bash-shell-change-the-color-of-my-shell-prompt-under-linux-or-unix/

Comment (sebastian-miclea) [reply lib/facter/framework/logging/logger.rb:8]: The test is quite bad, it doesn't actually test that the output has color. It tests that the output text is not formatted.
I'd rather fix the test.

File: lib/facter/framework/core/options/option_store.rb [hunk 2]
```diff
@@ -132,6 +132,7 @@ def show_legacy=(bool)
       end
 
       def trace=(bool)
+        @trace = bool
         Facter.trace(bool)
```
Comment (IrimieBogdan) [inline lib/facter/framework/core/options/option_store.rb:136]: We should remove `Facter.trace(bool)` and use the options in `facter.rb`

**Pull request merged**